extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts

```diff
@@ -0,0 +1,506 @@
+								throw err;
+							}
+							await this._postgresModel.refresh();
+						}
+					);
+
+					vscode.window.showInformationMessage(loc.instanceUpdated(this._postgresModel.info.name));
+
+					this.engineSettings = `'`;
+					this.engineSettingUpdates!.clear();
+					this.discardButton!.enabled = false;
+
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.instanceUpdateFailed(this._postgresModel.info.name, error));
+				}
+			}));
+
+		// Discard
+		this.discardButton = this.modelView.modelBuilder.button().withProps({
+			label: loc.discardText,
+			iconPath: IconPathHelper.discard,
+			enabled: false
+		}).component();
+
+		this.disposables.push(
+			this.discardButton.onDidClick(async () => {
+				this.discardButton!.enabled = false;
+				try {
+					// TODO
+					// this.parametersTable.data = [];
+					this.engineSettingUpdates!.clear();
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.pageDiscardFailed(error));
+				} finally {
+					this.saveButton!.enabled = false;
+				}
+			}));
+
+		// Reset
+		this.resetButton = this.modelView.modelBuilder.button().withProps({
+			label: loc.resetAllToDefault,
+			iconPath: IconPathHelper.reset,
+			enabled: true
+		}).component();
+
+		this.disposables.push(
+			this.resetButton.onDidClick(async () => {
+				this.resetButton!.enabled = false;
+				this.discardButton!.enabled = false;
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: loc.updatingInstance(this._postgresModel.info.name),
+							cancellable: false
+						},
+						async (_progress, _token): Promise<void> => {
+							//all
+							// azdata arc postgres server edit -n <server group name> -e '' -re
+							try {
+								await this._azdataApi.azdata.arc.postgres.server.edit(
+									this._postgresModel.info.name, { engineSettings: `'' -re` });
+							} catch (err) {
+								// If an error occurs while resetting the instance then re-enable the reset button since
+								// the edit wasn't successfully applied
+								this.resetButton!.enabled = true;
+								throw err;
+							}
+							await this._postgresModel.refresh();
+						}
+
+					);
+				} catch (error) {
+					vscode.window.showErrorMessage(loc.refreshFailed(error));
+				}
+			}));
+
+		return this.modelView.modelBuilder.toolbarContainer().withToolbarItems([
+			{ component: this.saveButton },
+			{ component: this.discardButton },
+			{ component: this.resetButton }
+		]).component();
+	}
+
+	private initializeConnectButton() {
+
+		this.connectToServerButton = this.modelView.modelBuilder.button().withProps({
+			label: loc.connectToServer,
+			enabled: false,
+			CSSStyles: { 'max-width': '125px' }
+		}).component();
+
+		this.disposables.push(
+			this.connectToServerButton!.onDidClick(async () => {
+				this.connectToServerButton!.enabled = false;
+				if (!vscode.extensions.getExtension('microsoft.azuredatastudio-postgresql')) {
```

Comment:
Suggest making this extension name a constant since it's referenced multiple times